app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -409,7 +417,12 @@ public void onExitFullScreen() {
             public void onDownloadStart(Download download) {
                 if (PackageManager.PERMISSION_GRANTED == ContextCompat.checkSelfPermission(getContext(), Manifest.permission.WRITE_EXTERNAL_STORAGE)) {
                     // We do have the permission to write to the external storage. Proceed with the download.
-                    queueDownload(download);
+                    if (download.getDestinationDirectory().equals(Environment.DIRECTORY_DOWNLOADS)) {
```

Comment:
This is my work-around to differentiate direct downloads that need this download dialog from long press on image downloads which have their own dialog.